dropshot/src/server.rs

```diff
@@ -578,15 +615,78 @@ impl<C: ServerContext> HttpServer<C> {
     }
 
     /**
-     * Signals the currently running server to stop and waits for it to exit.
+     * Return the result of registering the server's DTrace USDT probes.
+     *
+     * See [`ProbeRegistration`] for details.
      */
-    pub async fn close(mut self) -> Result<(), String> {
-        self.close_channel
+    pub fn probe_registration(&self) -> &ProbeRegistration {
+        &self.probe_registration
+    }
+}
+
+impl<C: ServerContext, Closer> HttpServer<C, Closer, Joinable> {
+    /// Separates the [HttpServer] from the [Joinable] object which is
+    /// capable of joining it.
+    pub fn get_joiner(self) -> (HttpServer<C, Closer, Unjoinable>, Joinable) {
+        let HttpServer {
+            probe_registration,
+            app_state,
+            local_addr,
+            closer,
+            joiner,
+        } = self;
+
+        (
+            HttpServer {
+                probe_registration,
+                app_state,
+                local_addr,
+                closer,
+                joiner: Unjoinable {},
+            },
+            joiner,
+        )
+    }
+}
+
+impl<C: ServerContext, Joiner> HttpServer<C, Closeable, Joiner> {
+    /// Separates the [HttpServer] from the [Closeable] object which is
+    /// capable of closing it.
+    pub fn get_closer(self) -> (HttpServer<C, Uncloseable, Joiner>, Closeable) {
```

Comment:
```suggestion
    /// Returns a handle that can be used to close the server
    pub fn closer(self) -> (HttpServer<C, Uncloseable, Joiner>, Closeable) {
```

Reply:
I opted to use the name `take_close_handle`, as I think it's more descriptive. It's a `take` operation because it changes the server, and it's more readable than `closer`.